src/CsharpClient/QuixStreams.Streaming/StreamConsumer.cs

```diff
@@ -135,6 +139,20 @@ private void OnEventDefinitionsReceived(IStreamPipeline streamPipeline, QuixStre
             this.OnEventDefinitionsChanged?.Invoke(this, obj);
         }
 
+        private void OnRawDataReceived(IStreamPipeline streamPipeline, byte[] bytes)
+        {
+            this.logger.LogTrace("StreamConsumer: OnRawDataReceived");
+            var ev = new EventDataRaw
+            {
+                Timestamp = DateTime.UtcNow.ToUnixNanoseconds() ,
```

Comment:
We need to use kafka message timestamp in this case, must be the same value regardless of how many times we read the same event. Preference would be that the Kafka specific logic only lives where kafka specific code currently does and not introduce at new places.

We want to disentangle ourselves from kafka in future for more supported brokers.


You might be able to get the required information in OnStreamPackageReceived handler, but that would also require polluting this class with Kafka specific logic.